src/upload-lighthouse/upload-to-s3.ts

```diff
@@ -0,0 +1,44 @@
+import * as github from '@actions/github';
+import {
+  ObjectCannedACL,
+  PutObjectCommand,
+  PutObjectCommandOutput,
+  S3Client,
+} from '@aws-sdk/client-s3';
+import { SortedRuns } from './types';
+
+// Html reports will be uploaded to S3
+export async function uploadHtmlToS3(
+  { htmlRuns, url }: SortedRuns,
+  type: 'mobile' | 'desktop',
+): Promise<PutObjectCommandOutput[]> {
+  const AWS_BUCKET = 'docs-lighthouse';
+  const commitHash = github.context.sha;
+  const branch = process.env.BRANCH_NAME || '';
+  const client = new S3Client();
+
+  const reportType = branch === 'main' ? 'main_reports' : 'pr_reports';
+
+  let cleanedUrl = url.replace('http://localhost:9000/', '');
+  if (cleanedUrl.endsWith('?desktop')) cleanedUrl = cleanedUrl.slice(0, -8);
+  cleanedUrl = cleanedUrl.split(/\/\/|\//).join('-');
```

Comment:
[nit] can we better construct the url so we don't join by `-` so that its directory path can be navigated by the url path.

also would remove the ?desktop and ?mobile the same way we do with localhost? perhaps good to make a separate util to derivePathFromReport?

Reply:
You know what, I think you're right here. I had it in my mind that it would be too many nested directories for no reason, but idk that's just how S3 is and when you have the choice to manipulate or not, I should just not. Sounds good to me!